README.md

```diff
@@ -19,6 +19,137 @@ it, simply add the following line to your Podfile:
+
+The `BaseViewController` class extends from `UIViewController` and adds useful, common utility that we have found ourselves adding in most, if not all, of our view controller classes.
+It does some additional set up in `viewDidLoad` to provide your view controller with keyboard notifications.  If overriding, be sure to call `super.viewDidLoad` in your subclass.
+
+### Example usage:
+
+```swift
+import MetovaBase
+
+class MyViewController: BaseViewController {
+
+    override func viewDidLoad() {
+    
+    	// this call registers your VC for keyboard notifications
+    	super.viewDidLoad() 
+    	
+    	// my custom setup here
+    }
+}
+```
+
+Additionally, `BaseViewController` unregisters your view controller from receiving keyboard notifications in its `deinit()` method.
+
+## KeyboardNotificationDelegate
+
+This protocol contains a collection of methods which `BaseViewController` subclasses (or extensions) can implement in order to hear about keyboard notifications. As already stated, you do not need to register or unregister for these notifications in your subclass or extension--you simply need to implement these methods.
+
+`BaseViewController` uses the `respondsToSelector:` method on itself to determine whether or not it should register for these notifications. 
+ As such, if implementing these in Swift, you must mark them as either `@objc` or `dynamic`.  If not, they are not implemented as selectors and `respondsToSelector:` returns `false`.
+ 
+### Methods
+ 
+ ```swift
+optional func keyboardWillShow(notification: Notification)
+ ```
+ 
+ Called when the [`UIKeyboardWillShowNotification`](https://developer.apple.com/reference/foundation/nsnotification.name/1621576-uikeyboardwillshow) is fired.
+ 
+ ```swift
+ optional func keyboardDidShow(notification: Notification)
+ ```
+ 
+ Called when the [`UIKeyboardDidShowNotification`](https://developer.apple.com/reference/foundation/nsnotification.name/1621602-uikeyboarddidshow) is fired.
+ 
+ ```swift
+ optional func keyboardWillChangeFrame(notification: Notification)
+ ```
+ 
+ Called when the [`UIKeyboardWillChangeFrameNotification`](https://developer.apple.com/reference/foundation/nsnotification.name/1621623-uikeyboardwillchangeframe) is fired.
+ 
+ ``` swift
+ optional func keyboardDidChangeFrame(notification: Notification)
+ ```
+ 
+ Called when the [`UIKeyboardDidChangeFrameNotification`](https://developer.apple.com/reference/foundation/nsnotification.name/1621619-uikeyboarddidchangeframe) is fired.
+ 
+ ```swift
+ optional func keyboardWillHide(notification: Notification)
+ ```
+ 
+ Called when the [`UIKeyboardWillHideNotification`](https://developer.apple.com/reference/foundation/nsnotification.name/1621606-uikeyboardwillhide) is fired.
+ 
+ ```swift
+ optional func keyboardDidHide(notification: Notification)
+ ```
+ 
+ Called when the [`UIKeyboardDidHideNotification`](https://developer.apple.com/reference/foundation/nsnotification.name/1621579-uikeyboarddidhide) is fired.
```

Comment:
Just to clarify, I'm saying we could structure it kind of like they do in [PureLayout's Readme](https://github.com/PureLayout/PureLayout#uiviewnsview).